File: rust/TypeScript/RustParserBase.ts
import { Parser, TokenStream } from 'antlr4';
import RustParser from './RustParser.js';

export default abstract class RustParserBase extends Parser {
    constructor(input: TokenStream) {
        super(input);
    }

    NextGT(): boolean {
        return this._input.LA(1) === RustParser.GT;
    }

    NextLT(): boolean {
        return this._input.LA(1) === RustParser.LT;
    }
}

File: rust/Dart/RustParserBase.dart
import 'package:antlr4/antlr4.dart';
import 'dart:io';
import 'dart:core';
import 'dart:convert';
import 'dart:collection';
import 'RustLexer.dart';
import 'RustParser.dart';

abstract class RustParserBase extends Parser {

    RustParserBase(TokenStream input) : super(input)
    {
    }

    bool NextGT() {
        return inputStream.LA(1) == RustParser.TOKEN_GT;
    }

    bool NextLT() {
        return inputStream.LA(1) == RustParser.TOKEN_LT;
    }
}